moveit_core/trajectory_processing/include/moveit/trajectory_processing/time_optimal_trajectory_generation.h

```diff
@@ -214,6 +246,15 @@ class TimeOptimalTrajectoryGeneration : public TimeParameterization
    */
   bool hasMixedJointTypes(const moveit::core::JointModelGroup* group) const;
 
+  /**
+   * @brief Check if the max scaling factor is valid and if not, set it to the specified default value
+   * \param[in,out] scaling_factor The default scaling factor that should be applied
+   * if the `max_scaling_factor` is invalid
+   * \param max_scaling_factor The desired maximum scaling factor to apply to the velocity or acceleration limits
+   * \param limit_type Whether the velocity or acceleration scaling factor is being verified
+   */
+  void verifyScalingFactor(double& scaling_factor, const double max_scaling_factor, const LimitType limit_type) const;
```

Comment:
I'm suggesting some small renaming to make this easier to understand.

It's confusing that `scaling_factor` is both an input and an output. Does it need to be an input? Can't we just default to 1.0 inside the function?

```suggestion
   * @brief Check if the requested scaling factor is valid and if not, set it to the specified default value
   * if `requested_scaling_factor` is invalid
   * \param requested_scaling_factor The desired maximum scaling factor to apply to the velocity or acceleration limits
   * \param limit_type Whether the velocity or acceleration scaling factor is being verified
   * \return The user requested scaling factor, if it is valid. Otherwise, return 1.0.
      */
  double verifyScalingFactor(const double requested_scaling_factor, const LimitType limit_type) const;
```